e2e/cypress/integration/keyboard_shortcuts/keyboard_shortcuts_3_spec.js

```diff
@@ -0,0 +1,102 @@
+                }
+            });
+        });
+    });
+
+    it('MM-T1575 - Ability to Switch Teams', () => {
+        for (let index = 0; index < count; index++) {
+            // # Verify that we've switched to the correct team
+            cy.uiGetLHSHeader().findByText(teams[count - index - 1].display_name);
+
+            // # Verify that we've switched to the correct channel
+            cy.get('#channelHeaderTitle').should('be.visible').should('contain', channels[count - index - 1].display_name);
+
+            // # Press CTRL/CMD+SHIFT+UP
+            if (isMac()) {
+                cy.get('body').type('{cmd}{option}', {release: false}).type('{uparrow}').type('{cmd}{option}', {release: true});
+            } else {
+                cy.get('body').type('{ctrl}{shift}', {release: false}).type('{uparrow}').type('{ctrl}{shift}', {release: true});
+            }
+        }
+
+        for (let index = 0; index < count; index++) {
+            // # Press CTRL/CMD+SHIFT+DOWN
+            if (isMac()) {
+                cy.get('body').type('{cmd}{option}', {release: false}).type('{downarrow}').type('{cmd}{option}', {release: true});
+            } else {
+                cy.get('body').type('{ctrl}{shift}', {release: false}).type('{downarrow}').type('{ctrl}{shift}', {release: true});
+            }
+
+            // # Verify that we've switched to the correct team
+            cy.uiGetLHSHeader().findByText(teams[index].display_name);
+
+            // # Verify that we've switched to the correct channel
+            cy.get('#channelHeaderTitle').should('be.visible').should('contain', channels[index].display_name);
+        }
+
+        for (let index = 1; index <= count; index++) {
+            // # Press CTRL/CMD+SHIFT+index
+            if (isMac()) {
+                cy.get('body').type('{cmd}{option}', {release: false}).type(String(index)).type('{cmd}{option}', {release: true});
+            } else {
+                cy.get('body').type('{ctrl}{shift}', {release: false}).type(String(index)).type('{ctrl}{shift}', {release: true});
+            }
+
+            // # Verify that we've switched to the correct team
+            cy.uiGetLHSHeader().findByText(teams[index - 1].display_name);
+
+            // # Verify that we've switched to the correct channel
+            cy.get('#channelHeaderTitle').should('be.visible').should('contain', channels[index - 1].display_name);
+        }
+
+        // # Verify keyboard shortcuts in Keyboard Shortcuts modal
+        cy.uiOpenHelpMenu('Keyboard shortcuts');
+        const name = isMac() ? /Keyboard Shortcuts ⌘ \// : /Keyboard Shortcuts Ctrl \//;
+        cy.findByRole('dialog', {name}).within(() => {
+            cy.findByText('Navigation').should('be.visible');
+            cy.findByText('Previous team:').should('be.visible');
+            cy.findAllByText('Next team:').should('be.visible');
+            cy.findAllByText('Switch to a specific team:').should('be.visible');
+        });
+    });
+});
```

Comment:
Extracted into a separate spec file since it requires complicated test data and takes long to complete the test.